scripts/kconfig/kconfigfunctions.py

```diff
@@ -11,11 +11,15 @@
 
 doc_mode = os.environ.get('KCONFIG_DOC_MODE') == "1"
 
-# The env var 'GENERATED_DTS_BOARD_CONF' must be set
-GENERATED_DTS_BOARD_CONF = os.environ['GENERATED_DTS_BOARD_CONF']
 
 dt_defines = {}
-if (not doc_mode) and os.path.isfile(GENERATED_DTS_BOARD_CONF):
+if (not doc_mode):
+    # The env var 'GENERATED_DTS_BOARD_CONF' must be set unless we are in
+    # doc mode
+    GENERATED_DTS_BOARD_CONF = os.environ['GENERATED_DTS_BOARD_CONF']
+    if not os.path.isfile(GENERATED_DTS_BOARD_CONF):
+         raise RuntimeError('Invalid path to GENERATED_DTS_BOARD_CONF: '
+                                            + GENERATED_DTS_BOARD_CONF)
```

Comment:
Maybe this is subjective, but does this RuntimeError exception really give a better error message than the FileNotFoundError exception that 'open' would give you?

Reply:
I agree, @ulfalizer said the same so I will revert this